### PR TITLE
Release 0.6.0

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,6 +68,18 @@ function(determine_version OUTPUT_VARIABLE)
     message(FATAL_ERROR "Failed to parse version number: ${GIT_VERSION}")
   endif()
 
+  if(APPLE OR MSVC)
+    string(REGEX MATCH "\\+full$" FULL_SUFFIX_MATCH ${NEW_VERSION})
+    if(NOT FULL_SUFFIX_MATCH)
+      set(NEW_VERSION ${NEW_VERSION}.full)
```
Comment (ricab) [inline CMakeLists.txt]: Shouldn't this be `+full` instead?

Comment (Saviq) [reply CMakeLists.txt]: No, the only two cases I'm expecting here are:
```
v0.7.0-dev.123+g456abc
v0.7.0+full
```
So we need to append `.full` to end up with dot-separated metadata.

Comment (ricab) [reply CMakeLists.txt]: OK thanks

**Comment by townsend2010** [comment]
Hmm, I think I need to make sure all the ducks are in a row before this merges.

bors r-

**Comment by townsend2010** [comment]
Ok, I think it's good.

bors r+

**Comment by townsend2010** [comment]
bors r=ricab,townsend2010

**Comment by townsend2010** [comment]
bors r=ricab,townsend2010

**Pull request merged**